cidc_api/gcloud_client.py

```diff
@@ -33,21 +33,33 @@ def _iam_id(user_email: str) -> str:
 
 
 def upload_xlsx_to_gcs(
-    trial_id: str, template_category: str, template_type: str, filebytes: BinaryIO
-) -> str:
+    trial_id: str,
+    template_category: str,
+    template_type: str,
+    filebytes: BinaryIO,
+    upload_moment: str,
+):
     """
     Upload an xlsx template file to GCS, returning the object URI.
     
     `template_category` is either "manifests" or "assays".
     `template_type` is an assay or manifest type, like "wes" or "pbmc" respectively.
+
+    Returns:
+        arg1: GCS blob object
     """
-    upload_moment = datetime.datetime.now()
-    blob_name = f"xlsx/{trial_id}/{template_category}/{template_type}/{upload_moment}"
+    blob_name = (
+        f"{trial_id}/xlsx/{template_category}/{template_type}/{upload_moment}.xlsx"
+    )
 
-    bucket: storage.Bucket = _get_bucket(GOOGLE_UPLOAD_BUCKET)
-    blob = bucket.blob(blob_name)
+    upload_bucket: storage.Bucket = _get_bucket(GOOGLE_UPLOAD_BUCKET)
+    blob = upload_bucket.blob(blob_name)
     blob.upload_from_file(filebytes)
-    return blob_name
+
+    data_bucket = _get_bucket(GOOGLE_DATA_BUCKET)
+    final_object = upload_bucket.copy_blob(blob, data_bucket)
+
+    return blob
```

Comment:
Should we return `final_object` here instead? Only might make a difference for the `time_created` attribute.